packages/@vue/cli-service/lib/Service.js

```diff
@@ -178,14 +178,14 @@ module.exports = class Service {
           ) {
             let apply = () => {}
             try {
-              apply = require(id)
+              apply = require(resolveModule(id, this.pkgContext))
             } catch (e) {
               warn(`Optional dependency ${id} is not installed.`)
             }
 
             return { id, apply }
           } else {
-            return idToPlugin(id)
+            return idToPlugin(id, resolveModule(id, this.pkgContext))
```

Comment:
`idToPlugin(resolveModule(id, this.pkgContext))` is better for me. And we don't need to modify the `idToPlugin` function.

Reply:
I can't do that as it would change the `id` in the object

Reply:
I think I ignored something about 'id', will check it later